src/service/google-drive-api-service.ts

```diff
@@ -0,0 +1,212 @@
+    }
+    const monthFolderId = await getFolderId(year, month);
+    if (!monthFolderId) return [];
+    const files: File[] = (await drive.files.list({
+      q: `'${monthFolderId}' in parents and mimeType = '${mimeType}'`,
+      fields: "files(id, name, mimeType)"
+    })).data?.files;
+    return files;
+  } catch (error) {
+    console.error(`Error retrieving files for year: ${year}, month: ${month}`, error);
+  }
+}
+
+/**
+ * Retrieves metadata for a specified file ID
+ * 
+ * @param id file ID
+ * @returns file's metadata
+ */
+export const getFile = async (id: string): Promise<File | undefined> => {
+  try {
+    const drive = await getDriveService();
+    const file = (await drive.files.get({
+      fileId: id,
+      fields: "id, name, mimeType",
+    }))?.data;
+    return file;
+  } catch (error) {
+    console.error(`Error retrieving file with ID: ${id}`, error);
+  }
+}
+
+/**
+ * Gets text content of a specified file in plain text format
+ * 
+ * @param file File metadata
+ * @returns file's content
+ */
+export const getFileText = async (file: File): Promise<string> => {
+  try {
+    const drive = await getDriveService();
+    const text = (await drive.files.export({
+      fileId: file.id,
+      mimeType: "text/plain"
+    })).data;
+    return text as string;
+  } catch (error) {
+    console.error(`Error retrieving text content of file: ${file.id}`, error);
+  }
+}
+
+/**
+ * Retrieves the binary content of a single PDF file 
+ * 
+ * @param file File object
+ * @param usedDrive Drive Instance
+ * @returns PDF file objects
+ */
+export const getFileContentPdf = async (file: File): Promise<PdfFile> => {
+  const drive = await getDriveService();
+  try {
+    const response = await drive.files.get({
+      fileId: file.id,
+      alt: "media",
+    },
+    { responseType: "stream"}
+    )
+    if (!response) {
+      console.error(`Failed to fetch file content: ${response.status} - ${response.statusText}`);
+    }
+    const pdfBuffer = await streamToBuffer(response.data);
+    return {
+      id: file.id,
+      name: file.name,
+      content: pdfBuffer
+    }
+  } catch (error) {
+    console.error("Error loading the PDF:", error);
+  }
+}
+  
+/**
+ * Creates a new PDF file from binary content and stores it in a specified folder
+ * 
+ * @param pdfFile PDF file object
+ * @param folderId folder ID
+ * @returns ID of files
+ */
+export const createPdfFile = async (pdfFile: PdfFile, folderId: string) => {
+  try {
+    const drive = await getDriveService();
+    const response = await drive.files.create({
+      requestBody: {
+        name: pdfFile.name,
+        mimeType: "application/pdf",
+        parents: [folderId]
+      },
+      media: {
+        mimeType: "application/pdf",
+        body: Readable.from(pdfFile.content)
+      }
+    });
+    return response.data.id;
+  } catch (error) {
+    console.error(`Failed to create PDF file: ${pdfFile.name} in folder ID: ${folderId}`, error);
+  }
```

Comment:
My understanding was Antti did not want anything uploading to the google drive, this includes the translations. I will check this with him before approving